comments.qml

```diff
@@ -1,55 +1,108 @@
+    description : qsTr("This plugin adds comments to your score")
+    pluginType : "dialog"
     //requiresScore: true // needs MuseScore > 2.0.3
-    pluginType: "dialog"
 
-    id:window
-    width:  400; height: 300;
-    onRun: {
-        if (!curScore)
+
+    onRun : {
+        if (!curScore) {
             Qt.quit();
+        } else {
+            window.visible = true;
+        }
+    }
+
+    Window {
+        id : window
+        width : 400;
+        height : 300;
+        visible : false;
+		
+        Settings {
+            id: settings
+            category : "pluginSettings"
+            property string metrics : ""
         }
 
-    Label {
-        id: textLabel
-        wrapMode: Text.WordWrap
-        text: qsTr("Add your comments")
-        font.pointSize:12
-        anchors.left: window.left
-        anchors.top: window.top
-        anchors.leftMargin: 10
-        anchors.topMargin: 10
+        Label {
+            id : textLabel
+            wrapMode : Text.WordWrap
+            text : qsTr("Add your comments")
+            font.pointSize : 12
+            anchors.left : window.left
+            anchors.top : window.top
+            anchors.leftMargin : 10
+            anchors.topMargin : 10
         }
 
-    TextArea {
-        id:abcText
-        anchors.top: textLabel.bottom
-        anchors.left: window.left
-        anchors.right: window.right
-        anchors.bottom: window.bottom
-        anchors.topMargin: 10
-        anchors.bottomMargin: 10
-        anchors.leftMargin: 10
-        anchors.rightMargin: 10
-        width:parent.width
-        height:400
-        focus: true
-        wrapMode: TextEdit.WrapAnywhere
-        textFormat: TextEdit.PlainText
-        Keys.onPressed: {
-            if (!curScore || event.key == Qt.Key_Escape)
-                Qt.quit()
-            curScore.setMetaTag("comments", text)
+        TextArea {
+            id : abcText
+            anchors.top : textLabel.bottom
+            anchors.left : window.left
+            anchors.right : window.right
+            anchors.bottom : window.bottom
+            anchors.topMargin : 10
+            anchors.bottomMargin : 10
+            anchors.leftMargin : 10
+            anchors.rightMargin : 10
+            width : parent.width
+            height : 400
+            focus : true
+            wrapMode : TextEdit.WrapAnywhere
+            textFormat : TextEdit.PlainText
+            Keys.onPressed : {
+                if (event.key == Qt.Key_Escape) {
+                    if (curScore) {
+                        curScore.setMetaTag("comments", text)
+                        var metrics = {
+                            x : window.x,
+                            y : window.y,
+                            width : window.width,
+                            height : window.height
+                        }
+                        settings.metrics =  JSON.stringify(metrics);
+                    }
+                    window.close();
```

Comment:
Do we really need to do all the above? I guess window.close will call onClosing, that will do exactly the same?